pylint/checkers/variables.py

```diff
@@ -1372,23 +1372,21 @@ def _check_consumer(
                 #   class A:
                 #      x = lambda attr: f + attr
                 #      f = 42
-                if isinstance(frame, nodes.ClassDef) and node.name in frame.locals:
-                    if isinstance(node.parent, nodes.Arguments):
-                        if stmt.fromlineno <= defstmt.fromlineno:
-                            # Doing the following is fine:
-                            #   class A:
-                            #      x = 42
-                            #      y = lambda attr=x: attr
-                            self.add_message(
-                                "used-before-assignment",
-                                args=node.name,
-                                node=node,
-                            )
-                    else:
-                        self.add_message(
-                            "undefined-variable", args=node.name, node=node
-                        )
-                        return (VariableVisitConsumerAction.CONSUME, found_nodes)
+                # We check lineno because doing the following is fine:
+                #   class A:
+                #      x = 42
+                #      y = lambda attr: x + attr
+                if (
+                    isinstance(frame, nodes.ClassDef)
+                    and node.name in frame.locals
+                    and stmt.fromlineno <= defstmt.fromlineno
+                ):
+                    self.add_message(
+                        "used-before-assignment",
+                        args=node.name,
+                        node=node,
+                    )
```

Comment:
```suggestion
                    self.add_message(
                        "used-before-assignment",
                        args=node.name,
                        node=node
                    )
```
Trying to compact the code with black.